main.tf

```diff
@@ -99,6 +99,17 @@ resource "google_project" "project" {
   app_engine = "${local.app_engine_config["${local.app_engine_enabled ? "enabled" : "disabled"}"]}"
 }
 
+/******************************************
+  Project lien
+ *****************************************/
+resource "google_resource_manager_lien" "lien" {
+  count        = "${var.lien ? 1 : 0}"
```

Comment:
Now that a string is being used, you need to check for the string explicitly. See how we handle random ID.

Reply:
When we check the `random_project_id` variable we don't do an explicit string comparison, as given by the following snippet from `main.tf:

```hcl
locals {
  temp_project_id        = "${var.random_project_id ? format("%s-%s",var.name,random_id.random_project_id_suffix.hex) : var.name}"
  # [...]
}
```

I've also tested the behavior in the terraform console:
```
> "${"true" ? 1 : 0}"
1
> "${"false" ? 1 : 0}"
0
> "${"shrug" ? 1 : 0}"
__builtin_StringToBool: strconv.ParseBool: parsing "shrug": invalid syntax in:

${"${"shrug" ? 1 : 0}"}
```

That behaves as expected, only allowing true or false and rejecting other strings.

If we do an explicit string comparison, we lose the validation of the input as a boolean:

```
> "${"true" == "true" ? 1 : 0 }"
1
> "${"false" == "true" ? 1 : 0 }"
0
> "${"shrug" == "true" ? 1 : 0 }"
0
```

Have I missed something with how we should handle booleans?

Reply:
The issue is we're relying on Terraform's implicit coercion from "true" to 1 which is not guaranteed.

https://www.terraform.io/docs/configuration/variables.html#booleans

Reply:
Mirroring @morgante's comment here. Once that's resolved have a 👍 from me.

Reply:
@adrienthebo Where did we land on this? Would like to merge soon.

Reply:
Per the earlier discussion, this comparison is valid but we should use `"true"` over `true` as the variable input. I've updated the use of the bare `true` in the tests and updated this PR.